### DISCUSSION TITLE
[Question]: NuGet missing packages in local dev environment

### DISCUSSION THREAD

**Discussion by pieter-kok** [microsoft/AL-Go#1592] [Q&A]
### Question

Currently, the missing dependencies for the AppSource apps we're dependent on are retrieved from the Microsoft NuGet feed during running the pipeline. That is a fantastic addition for us, a real unblocker.

Another part, that still seems to be blocked, is creating local development environments on docker.
I would have expected that creating a local dev environment with localDevEnv.ps1 would have resulted in a development environment with those NuGet symbols included. That seems not to be the case. Is that behavior as expected?

How should we create local dev environments in the specific scenarios where we are dependent on an AppSource app? Is there f.i. a work-around to get this app from the NuGet feed and push it into docker environment during the creating the environment by localDevEnv.ps1?

**Comment by frottke** [comment]
> I would have expected that creating a local dev environment with localDevEnv.ps1 would have resulted in a development environment with those NuGet symbols included. That seems not to be the case. Is that behavior as expected?

We would have expected the same, would be great if this works :)

**Comment by frottke** [comment]
Hi @freddydk, I was wondering if you could provide an update on this issue. 

We’re looking into NuGet as a possible alternative to `appDependencyProbingPaths`, given that `appDependencyProbingPaths `doesn’t allow reading GitHub Secrets. However, we’re concerned that NuGet might have a similar limitation, especially due to the `authTokenSecret `in `NuGetAuthContext`. If that’s the case, any feedback would help us focus on exploring other solutions on how to access private nuget feeds .

**Comment by freddydk** [comment]
GitHubPackages should work.
Your user would obviously need to have access to the github packages - but then it should be able to download them

**Comment by frottke** [comment]
Usually our users do not have access to the private repository (ISV products) of other organisations.  We would get provided with a PAT from the external organisation.  

New Question might be now: How to access the private nuget or GitHubPackages feed of an other organisation that will not provide access (justifiably) to their private repository and will only provide access via a PAT to reduce the permissions to a minimum (reading packages). 

We are currently in the same situation, we would have to give many users access to our repository as external collaborators, which also generates costs

**Comment by jonaswre** [comment]
@freddydk this is also why only providing symbols is not sufficient. 


But as for a solution to your problem. 

We use the https://github.com/BusinessCentralApps/GenerateBcNuGet Repo to generate a Nuget feed thats contains the runtimes which then can be shared by inviting our partners to that organisation. This is currently not a perfect solution and be mainly use it to distribute runtimes for onprem instaltions. The Main reason why its not the best solution is because in the form its currently public it needs some manual work for each release, i've made some changes to improve the workflow but there is still full AL-Go Integration missing.

**Comment by jwikman** [comment]
@aholstrup1, do you know if there are any plans of supporting NuGet feeds when using `localDevEnv.ps1`?

I guess that authentication is the most challenging part here...?
But in my scenario, the feeds are in organizations where the user is authenticated, so `gh auth token` should return a token that could be used for authentication.
